src/payment/strategies/moneris/moneris-payment-strategy.ts

```diff
@@ -89,29 +141,74 @@ export default class MonerisPaymentStrategy implements PaymentStrategy {
         if (nonce !== undefined) {
             return this._store.dispatch(this._paymentActionCreator.submitPayment({
                 methodId: payment.methodId,
-                paymentData: { nonce },
+                paymentData: { nonce, shouldSaveInstrument, shouldSetAsDefaultInstrument },
             }));
         }
 
         return this._store.getState();
     }
 
-    finalize(): Promise<InternalCheckoutSelectors> {
-        return Promise.reject(new OrderFinalizationNotRequiredError());
-    }
+    protected async executeWithVaulted(payment: OrderPaymentRequestBody): Promise<InternalCheckoutSelectors> {
```

Comment:
Why is this a protected method and not a private? and can we please add a _ as a prefix.